### ISSUE TITLE
Wrong Return URL after Payment Due to this Plugin

### ISSUE THREAD

**Issue by jannisf** [hyyan/woo-poly-integration#12]
I am using this plugin together with the "PostFinance E-Payment Gateway for WooCommerce" by mame dev. After finalizing a payment I got to see the "Oops, this page cannot be found" every time. I went through lengthy trouble shooting sessions with the developer mame dev until he found out that it is actually this "Hyyan WooCommerce Polylang Integration" plugin causing the return URL to be:
http://www.[domain].com/en/checkout//1769/order-received/1769/?key=wc_order_56212c7e09a10&v=d1a0eba20b32
instead of
http://www.[domain].com/en/checkout/order-received/1769/?key=wc_order_56212c7e09a10&v=d1a0eba20b32
The PostFinance plugin gets its return URL through the WooCommerce function "get_return_url()", which is apparently modified by this plugin in some mysterious way.
Can you please fix this bug so that this WooCommerce Polylang Integration plugin works seamlessly with all payment gateways? Thanks very much!

**Comment by theblackhole** [comment]
Hey @jannisf ,
It's probably too late for you but for it may be useful for the others.
After a long diagnostic, I fixed this issue by commenting [line 250 ($value) in blob/master/src/Hyyan/WPI/Endpoints.php](https://github.com/hyyan/woo-poly-integration/blob/master/src/Hyyan/WPI/Endpoints.php#L250) :

```php
$url = trailingslashit($permalink)
          . $endpoint
          . '/'
          //. $value
          . $query_string;
```

Now it generates the correct url :smile:

**Comment by TeemuSuoranta** [comment]
I made much more complicated solution because I couldn't find the exact cause. I made a pull request and my solution is working as well https://github.com/hyyan/woo-poly-integration/pull/25

**Comment by hyyan** [comment]
Fixed in version 0.27
